src/sources/gcp_pubsub.rs

```diff
@@ -264,16 +275,16 @@ impl PubsubSource {
         };
         let mut stream = stream.into_inner();
 
-        if let Some(credentials) = self.credentials.take() {
-            credentials.spawn_regenerate_token();
-        }
-
         let (finalizer, mut ack_stream) =
             Finalizer::maybe_new(self.acknowledgements, self.shutdown.clone());
 
         loop {
             tokio::select! {
                 _ = &mut self.shutdown => return false,
+                _ = &mut token_generator.next() => {
+                    info!("New authentication token generated, restarting stream.");
```

Comment:
I think this could probably be `debug` too. Most users won't care about it if everything is operating normally.

Reply:
I was going to ask about that, actually, forgot to put it in the description above. What do you think about the `info!` above in the main loop? https://github.com/vectordotdev/vector/blob/86b94ad037ff32911923831a9fabd5a4a3ef60c1/src/sources/gcp_pubsub.rs#L225

Reply:
For that one, I wonder if `run_once` should return the error (`Result<, _>`) and then it could just log one `warn!` log with the message and the fact that it is retrying? Currently it seems like it always returns `true` (i.e. retry) whenever an error occurs.